src/physics/initial_conditions/initial_condition_function.cpp

```diff
@@ -509,6 +676,7 @@ InitialConditionFactory<dim,nstate, real>::create_InitialConditionFunction(
         }
     } else if (flow_type == FlowCaseEnum::burgers_inviscid && param->use_energy==false) {
         if constexpr (dim==1 && nstate==1) return std::make_shared<InitialConditionFunction_BurgersInviscid<dim,nstate,real> > ();
+        if constexpr (dim==2 && nstate==2) return std::make_shared<InitialConditionFunction_BurgersInviscid<dim, nstate, real> >();
```

Comment:
```suggestion
        if constexpr (nstate==dim && dim<3) return std::make_shared<InitialConditionFunction_BurgersInviscid<dim, nstate, real> >();
```

Reply:
can delete the line above; this can also be applied in similar instances in this PR

Reply:
Modified the if statement to the given condition. Made the changes in tests.cpp as well but haven't committed the tests.cpp changes yet.